Modix.Bot/Behaviors/PromotionPollBehavior/PromotionDialogueBehavior.cs

```diff
@@ -0,0 +1,161 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Linq;
+using Discord;
+using Discord.WebSocket;
+using Microsoft.Extensions.Caching.Memory;
+using Modix.Common.Messaging;
+using Modix.Data.Models.Core;
+using Modix.Data.Models.Promotions;
+using Modix.Services.Core;
+using Modix.Services.Images;
+using Modix.Services.Promotions;
+using Modix.Services.Utilities;
+
+namespace Modix.Bot.Behaviors
+{
+    public class PromotionDialogueBehavior :
+        INotificationHandler<ReactionAddedNotification>,
+        INotificationHandler<PromotionActionCreatedNotification>
+    {
+        private readonly DateTime _utcNow = DateTime.UtcNow;
```

Comment:
Do we have an `ISystemClock` for getting `UtcNow` or is that just in my own projects?